### PR TITLE
Various CMake improvements

### PR DESCRIPTION
**Pull request by lzaoral**
* Handle static linking in CMake so that we don't overwrite default flags in Fedora and RHEL. 
* Use `GNUInstallDirs` module to handle recommended install directories (e.g. `lib` vs. `lib64`) for the given distro.
* Let CMake handle hard-coded paths in `csexec-loader` and `cswrap`.
* Generally modernise CMake to use features present in 3.4.
* Make `csexec`'s CMake platform independent in preprataion for #9.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -86,10 +86,10 @@ if(LD_LINUX_SO)
 
     # readlink() wrapper
     add_library(csexec-preload SHARED csexec-preload.c)
-    set_target_properties(csexec-preload PROPERTIES LINK_FLAGS "-nostdlib")
+    set_target_properties(csexec-preload PROPERTIES LINK_FLAGS "-nostdlib -Wl,-Bstatic -l${CMAKE_DL_LIBS} -pthread")
     target_compile_definitions(csexec-preload PRIVATE LD_LINUX_SO="${LD_LINUX_SO}")
-    target_link_libraries(csexec-preload PRIVATE ${CMAKE_DL_LIBS}
-                                                 Threads::Threads)
+    #    target_link_libraries(csexec-preload PRIVATE ${CMAKE_DL_LIBS}.a
+    #                                             Threads::Threads)
```
Comment (lzaoral) [inline src/CMakeLists.txt:92]: The last fixup fixes that `libcsexec-preload` was not linked statically but to be honest, I don't like this solution at all.

Comment (kdudka) [reply src/CMakeLists.txt:92]: I do not think we need `libcsexec-preload.so` to be linked statically.  We can reconsider it later on when we have a real use case for it.

**Review by kdudka — APPROVED** [review]

File: src/CMakeLists.txt (hunk 1 above)
Comment (lzaoral) [reply src/CMakeLists.txt:92]: Ok, Iet's link it dynamically, then. That should make things much simpler.

**Comment by lzaoral** [comment]
I can also confirm that these changes fix the linking error in `epel-8-aarch64` build in #9.

**Review by kdudka — APPROVED** [review]
Looks perfect.

**Comment by lzaoral** [comment]
Thanks a lot, @kdudka! Merging.

**Pull request merged**